pkg/tools/cli/loadtest/service/loader.go

```diff
@@ -88,48 +99,64 @@ func (l *loader) Prepare(context.Context) (err error) {
 	if err != nil {
 		return err
 	}
-	l.requests, err = l.requestsFunc(dataset)
+
+	switch l.operation {
+	case config.Insert, config.StreamInsert:
+		l.dataProvider, l.dataSize, err = insertRequestProvider(dataset, l.batchSize)
+	case config.Search, config.StreamSearch:
+		l.dataProvider, l.dataSize, err = searchRequestProvider(dataset)
+	}
```

Comment:
I think applying the comments below will make things simpler.
https://github.com/vdaas/vald/pull/552/files#r462742394

for example.

l. dataProvider, err = provider.DataFunc()
if err != nil {
    return nil, err
}

Reply:
If this PR includes fix about provider, I think it is too large.

Reply:
ok, It seems to be good in the direction of responding with another PR.

cc: @kpango